core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -155,15 +149,28 @@ protected void validate() {
       (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
       throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
     }
-    if (!format.equalsIgnoreCase("text")) {
-      try {
-        new Schema.Parser().parse(inputSchema);
-      } catch (Exception e){
-        throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+    Schema parsedSchema;
+    try {
+      parsedSchema = Schema.parseJson(schema);
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Unable to parse schema with error: " + e.getMessage(), e);
+    }
+
+    if (format.equalsIgnoreCase("text")) {
+      if (schema != null && !schema.isEmpty()) {
+        if (!(schema.contains("offset") && schema.contains("body"))) {
+          throw new IllegalArgumentException("Schema must contain fields for offset and body");
+        }
       }
-    } else if (inputSchema != null && !inputSchema.isEmpty()) {
-      if (!(inputSchema.contains("offset") && inputSchema.contains("body"))) {
-        throw new IllegalArgumentException("Schema must contain fields for offset and body");
+    }
+    if (pathField != null) {
+      if (schema.contains(pathField)) {
+        Schema.Field schemaPathField = parsedSchema.getField(pathField);
+        if (!(schemaPathField.getSchema().isSimpleOrNullableSimple())) {
+          throw new IllegalArgumentException("Value for " + pathField + " must be a string or nullable string");
+        }
+      } else {
+        throw new IllegalArgumentException("Schema must contain " + pathField);
```

Comment:
For readability, logic for handling a condition should be as close to the condition as possible. In this case, move things around to look like:
```
Schema.Field schemaPathField = parsedSchema.getField(pathField);
if (schemaPathField == null) {
  // throw condition
}
// check type
```